psutil/_psaix.py

```diff
@@ -415,7 +410,12 @@ def exe(self):
 
     @wrap_exceptions
     def cmdline(self):
-        return self._proc_name_and_args()[1].split(' ')
+        try:
+            return cext.proc_args(self.pid) or ['']
```

Comment:
This is what I did at first. Then I found out that it breaks backwards compatibility, and breaks other methods that rely on the list to not be empty.
The previous implementation used `.split(' ')`, so even in the case of empty string, the result would be a list with one empty string in it. I didn't want to change this behavior.
The method `exe` uses `cmdline()[0]` so if we would return an empty list, this function would raise an exception instead of returning an empty string itself.

Reply:
mmmm it seems exe() should be fixed then (I suppose in psutil/__init__.py but can’t check now :)).

Reply:
looks like we check if cmdline exists first so it should be safe:
https://github.com/giampaolo/psutil/blob/master/psutil/__init__.py#L758

Reply:
The exe implementation in `psutil.__init__` correctly checks `if cmdline ...` before trying to read it. AIX doesn't do that check. I can fix it, but what I'm afraid of is maybe breaking other applications that came to rely on `cmdline` never being empty. There are probably none, though. Do you think it's ok to "break backward compatibility" here?

Reply:
AIX also implicitly relies on that check (all platforms do) so we should be safe. I am not sure if there are apps (or unit tests) relying on cmdline being non-empty. If they do I would say they should not and hence should be fixed. The correct “null value” for an empty cmdline should be an empty list, the same way, for instance, the “null value” for net_if_addrs() is an empty dict.

Reply:
Now i wonder if there are platforms returning [“”]. I hope not... but that’s for another time.